src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,22 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
-  end
-
-  def general_state
-    build_state == :acceptable ? check_state : build_state
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.exists?
+    build_or_check_state = build_state || check_state
+    return build_or_check_state if build_or_check_state
```

Comment:
Why don't we move all of this into one method? Are we checking `.build_state` independently from `check_state` somewhere?

Reply:
I don't know if I understood @hennevogel, but isn't better to hide all the "details" on methods and when reading the "state" method, you don't have to wrap your head around the details, but a quick glance on the method names, should make clear when the states are set?

Reply:
I am also not sure what @hennevogel means... Maybe it goes in the same direction as https://github.com/openSUSE/open-build-service/pull/7262#discussion_r268508491 :thinking:

Reply:
I mean make `build_or_check_state` a method. Not two method and a variable.

Reply:
@hennevogel ah yes, and preferably we see all return returning a ```:value```

Reply:
I'm not sure if I understood.

@Ana06 suggested to get rid of the wrapper method `build_or_check_state` and not to call it twice (so, we assigned the condition to a variable).

@hennevogel, do you suggest to create the wrapper method again an call inside to the other two? Like this:

```
  def build_or_check_state                                                                               
    build_state || check_state                                                                                          
  end
```

Or do you propose that we have only one method with all the code inside? like:

```
  def build_or_check_state
    # build_state                                                                              
    return :building if building_repositories.present? || disabled_repositories.present?                 
    return :failed if broken_packages.present?                                                                  
    # check_state
    return :testing if missing_checks.present? || checks.pending.exists?                                 
    :failed if checks.failed.exists?                                                                     
  end 
```
(we thought about it but found it uglier, or semantically unclear)

@vpereira we can discuss later if we create a new method for each condition. Let's decide this before.

Reply:
the latter